spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/feign/TraceLoadBalancerFeignClient.java

```diff
@@ -95,6 +101,11 @@ public Response execute(Request request, Request.Options options) throws IOExcep
 		}
 	}
 
+	private boolean delegateIsALoadBalancer() {
+		return getDelegate() instanceof LoadBalancerFeignClient
```

Comment:
Ug. I guess there's nothing better. :man_shrugging:

Reply:
Yes, not sure we can do it any better way. In `master` we will only have to check against one class, at least.